pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -159,12 +160,25 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "contact_info") {
 		adminContact, techContact := toPBContactInfo(req.Gateway.GetContactInfo())
-		updateReq.AdministrativeContact = &packetbroker.ContactInfoValue{
-			Value: adminContact,
+
+		// Only replaces the administrative and technical contacts if no error happens.
+		adminID, err := s.fetchUserIDFromContact(
+			ctx, req.Gateway.GetAdministrativeContact(),
+			func(o *ttnpb.Organization) *ttnpb.OrganizationOrUserIdentifiers { return o.GetAdministrativeContact() },
+		)
+		if err == nil {
+			adminContact = &packetbroker.ContactInfo{Name: adminID.GetUserId(), Email: adminID.GetEmail()}
```

Comment:
Why are you using the identifiers email again ? We've already settled that it is empty and unused. You need to retrieve the `primary_email_address` of the `User` entity. The `UserIdentifiers.Email` is not usable.